libusb/os/windows_winusb.c

```diff
@@ -1480,6 +1480,110 @@ static int set_hid_interface(struct libusb_context *ctx, struct libusb_device *d
 	return LIBUSB_SUCCESS;
 }
 
+static int get_guid(struct libusb_context *ctx, char *dev_id, HDEVINFO *dev_info, SP_DEVINFO_DATA *dev_info_data,
+					int guid_number, GUID **if_guid)
+{
+	DWORD size, reg_type;
+	HKEY key;
+	char *guid_string, *new_guid_string;
+	char *guid;
+	LONG s;
+	size_t len;
+	int i;
+	int err = LIBUSB_SUCCESS;
+
+	if (!pSetupDiGetDeviceRegistryPropertyA(*dev_info, dev_info_data, SPDRP_DRIVER,
```

Comment:
Potentially we could leave this check in the caller if is kind of independent? Or is it more "efficient" to have it here? Maybe at least duplicate here the comment in the caller about missing driver check.

Reply:
You're right, it makes sense to leave this check in the caller because this function reads only from registry key that might be independent on the driver state.